benchmarks/oss.py

```diff
@@ -30,9 +30,12 @@ def dist_init(rank, world_size, backend):
     dist.init_process_group(backend=backend, init_method="tcp://localhost:29501", rank=rank, world_size=world_size)
 
 
-def get_problem(rank, data_size, batch_size, device):
-    # Standard RN101
-    model = resnet101(pretrained=False, progress=True).to(device)
+def get_problem(rank, data_size, batch_size, device, resnet: str):
+    # Select the desired model on the fly
+    print(f"Using {resnet} for benchmarking")
+    model = {"resnet18": resnet18, "resnet50": resnet50, "resnet101": resnet101, "resnet152": resnet152}[resnet](
```

Comment:
should we use importlib to import models directly from `torchvision.models`, that would avoid some glue code?

Reply:
ahh I didn't know that, would you have a pointer ?

Reply:
an example:

```
import importlib


m1 = importlib.import_module('example.submodule')
print(m1)

m2 = importlib.import_module('.submodule', package='example')
print(m2)

print(m1 is m2)

```

you can turn string into import modules I think.

Reply:
the models are not modules really, so it does not work OOB, but looking into the syntax